### PR TITLE
implementation for sha & ssha 1|256|512

### PR DESCRIPTION
**Pull request by elbotho**

### CODE REVIEW

File: hash/hash_comparator.go [hunk 1]
```diff
@@ -203,16 +171,43 @@ func CompareSSHA512(_ context.Context, password []byte, hash []byte) error {
...
+	// Extract the parameters, salt and derived key from the encoded password
+	pf, salt, hash, err := decodeSHA1Hash(string(hash))
+	if err != nil {
+		return err
+	}
+
+	r := strings.NewReplacer("{SALT}", string(salt), "{PASSWORD}", string(password))
+	arg := []byte(r.Replace(string(pf)))
+
+	// @hugo: why did you use hex.EncodeToString here before?
```
Comment (hugotiburtino) [inline hash/hash_comparator.go:188]: that was the hacky part. I couldn't find a quick way of comparing bytes, so I used strings instead.
We can delete it, if your implementation works

Comment (elbotho) [reply hash/hash_comparator.go:188]: It does not unfortunately ^^
I'll add a working test in a minute and then we can work on that :)

Comment (elbotho) [reply hash/hash_comparator.go]: Okay, should work now (see slack)

**Review by hugotiburtino — APPROVED** [review]
Beautiful!

**Pull request merged**